android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt

```diff
@@ -73,13 +73,14 @@ class UrlUtil {
                         }
                     }
                 } else if (service.equals(ENVOY_SERVICE_SNOWFLAKE)) {
+                    // handles both submitted and local urls for snowflake service
                     // extract domain from queries
                     val uri = URI(url)
                     val rawQuery = uri.rawQuery
                     val queries = rawQuery.split("&")
                     for (i in 0 until queries.size) {
                         val queryParts = queries[i].split("=")
-                        if (queryParts[0].equals("url")) {
+                        if (queryParts[0].equals("tunnel") || queryParts[0].equals("url")) {
```

Comment:
where is "tunnel" coming from?

Reply:
it's one of the query elements in the submitted url.  the value becomes the "url" query element in the local url used for snowflake